### PR TITLE
Adding minio-driver

### PR DESCRIPTION
**Pull request by sajanjswl**
Signed-off-by: Sajan Jaiswal <sjnjaiswal@gmail.com>

### CODE REVIEW

File: examples/sample-provisioner/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	var cancel context.CancelFunc
+
+	ctx, cancel = context.WithCancel(context.Background())
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, syscall.SIGSEGV)
+
+	go func() {
+		s := <-sigs
+		cancel()
+		panic(fmt.Sprintf("%s %s", s.String(), "Signal received. Exiting"))
```
Comment (wlan0) [inline examples/sample-provisioner/cmd/main.go:99]: No need to panic here

**Comment by rrati** [comment]
Part of #8

File: examples/sample-provisioner/driver/driverserver.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+func (ds *DriverServer) ProvisionerGetInfo(context.Context, *cosi.ProvisionerGetInfoRequest) (*cosi.ProvisionerGetInfoResponse, error) {
+	rsp := &cosi.ProvisionerGetInfoResponse{}
+
+	rsp.ProvisionerIdentity = fmt.Sprintf("%s-%s", ds.Name, ds.Version)
+	return rsp, nil
+}
+
+func (ds DriverServer) ProvisionerCreateBucket(ctx context.Context, req *cosi.ProvisionerCreateBucketRequest) (*cosi.ProvisionerCreateBucketResponse, error) {
+
```
Comment (wlan0) [inline examples/sample-provisioner/driver/driverserver.go:50]: This is empty?

**Comment by wlan0** [comment]
I dont like the idea of having an entirely different go dependency tree for a proejct under the same root. Let's discuss how to simplify this. cc @rrati

**Comment by rrati** [comment]
@wlan0 Agreed.  It was always intended to pull the example provisioner into the main build structure, we just didn't get there before the move.  This seems like a perfect time to correct that and not put the sample provisioner in its own tree.

**Comment by rrati** [comment]
/close

Done as part of #27

**Pull request closed without merging**